ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -168,12 +177,14 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
           Array.isArray(echartsPalette) && echartsPalette[0]
             ? (echartsPalette[0] as string)
             : muiTheme.palette.primary.main;
-        // Check which color palette was chosen and get appropriate color.
+
+        // Decide which palette to use based on total series returned.
+        // When series number exceeds number of colors in palette, use generative colors instead.
+        const paletteLength = Array.isArray(echartsPalette) ? echartsPalette.length : 4;
```

Comment:
One more fix coming soon! I'm planning to rename `echartsPalette` to `categoricalPalette` based on @cndonovan's [feedback here](https://github.com/perses/perses/pull/1088#discussion_r1175519897) on PR #1088. Otherwise, I think I've fixed the remaining comments on the previous PR and this one is officially ready for review now